### ISSUE TITLE
Feature request: Possible to introduce tf.complex32 datatype

### ISSUE THREAD

**Issue by beniroquai** [tensorflow/tensorflow#15576]
Just recently I figured out, that our GPU (TitanX 12GB) is not big enough of its memory capacities to process our data. I was wondering whether there are attempts to "quantize" the complex datatypes as well. 

I've seen that there are approaches to quantize float32 datatypes for example and was wondering if this is possible for real/imag types as well. 

Thanks a lot!

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by asimshankar** [comment]
Nope, there aren't any active efforts towards this AFAIK.

Closing this out since the feature request itself is a bit broad and there is nobody actively working on this.

**Comment by zaccharieramzi** [comment]
I know this issue has been closed for being too broad, but I am still wondering whether it should be considered for people trying to optimize their model.
In particular, [this section](https://www.tensorflow.org/guide/profiler#improve_device_performance) specifies that you should use half-precision to improve device performance, but it isn't possible with complex dtypes (which are needed for Fourier transforms for example).

I haven't looked into [mixed precision](https://www.tensorflow.org/guide/keras/mixed_precision) yet (to have all other operations performed with half-precision), but a complex32 dtype would simplify things.
